### PR TITLE
OpenVINO optimizations for Knowledge graphs embedding model (ConvE)

### PR DESCRIPTION
**Pull request by karkadad**
This notebook showcases optimizations that can be obtained using the Intel OpenVINO toolkit on a Knowledge Graph embeddings model (ConvE).

We showcase the following:
1. The trained PyTorch knowledge graph embeddings model conversion for OpenVINO inference.
2. Show that there were no accuracy differences post conversion.
3. Showcase the platform specific speedup in runtime performance that can be obtained through OpenVINO graph optimizations.

I have performed manual test and code quality checks (nbval and nbqa). I have made sure to follow the coding guidelines provided here: https://github.com/openvinotoolkit/openvino_notebooks/blob/173645d14740b0ceba9d9785552823bf8330ce67/CONTRIBUTING.md#coding-guidelines .

Research paper and code attribution are also provided and I've also included the model implementation license in the Jupyter notebook.

### CODE REVIEW

**Comment by karkadad** [comment]
@adrianboguszewski Please let me know who can review this. Thanks.

**Comment by karkadad** [comment]
@ryanloney Thanks for the great feedback, I have updated the README for a bit of background context on the model and the task and then included a couple of cells on how to do prediction on the model. I've also included cells showcasing how to do prediction using benchmark_app for this model. 

As for the Windows runs, looks like some of the nbval runs successfully completed on Windows, and there is a template from other notebooks on finding cl.exe included in the notebook, however I will double check to make there are no issues.

**Comment by karkadad** [comment]
@ryanloney I did additional testing on Windows, but did not see the cl.exe issue. The notebook was able to find cl.exe and add it PATH without additional user inputs. Could you provide more details on how to reproduce the issue?
I have also updated the notebook and README on how to interact with the model and also included instructions on running with benchmark app as requested, please review when you get a chance. Thanks!

**Comment by karkadad** [comment]
@ryanloney Please let me know if you had a chance to review this. Thanks

File: notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ie = Core()\n",
+    "ir_net = ie.read_model(model=fp32_onnx_path)\n",
+    "compiled_model = ie.compile_model(model=ir_net)\n",
+    "output_layer = compiled_model.output(0)\n",
```
Comment (Debskij) [inline notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb:388]:
```suggestion
    "input_layer_source = compiled_model.input('input.1')\n",
    "input_layer_relation = compiled_model.input('input.2')\n",
    "output_layer = compiled_model.output(0)\n",
```

File: notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+    "request = compiled_model.create_infer_request()\n",
+    "\n",
+    "ov_acc = 0.0\n",
+    "ov_inf_times = []\n",
+    "for i in range(num_test_samples):\n",
+    "    test_sample = triples_list[i]\n",
+    "    h, r, t = test_sample\n",
```
Comment (Debskij) [inline notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb:395]:
```suggestion
    "    source, relation, target = test_sample\n",
    "    model_inputs = {input_layer_source: source, input_layer_relation: relation}\n",
```

File: notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ie = Core()\n",
+    "ir_net = ie.read_model(model=fp32_onnx_path)\n",
+    "compiled_model = ie.compile_model(model=ir_net)\n",
+    "output_layer = compiled_model.output(0)\n",
+    "request = compiled_model.create_infer_request()\n",
```
Comment (Debskij) [inline notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb:389]:
```suggestion

```
Remove this line

File: notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+    "\n",
+    "ov_acc = 0.0\n",
+    "ov_inf_times = []\n",
+    "for i in range(num_test_samples):\n",
+    "    test_sample = triples_list[i]\n",
+    "    h, r, t = test_sample\n",
+    "    start_time = time.time()\n",
+    "    request.infer(inputs={'input.1': h, 'input.2': r})\n",
```
Comment (Debskij) [inline notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb:397]:
```suggestion
    "    result = compiled_model(model_inputs)[output_layer]\n",
```

File: notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+    "    end_time = time.time()\n",
+    "    result = request.get_output_tensor(output_layer.index).data\n",
```
Comment (Debskij) [inline notebooks/216-knowledge-graphs-conve/216-knowledge-graphs-conve.ipynb:399]:
```suggestion

```
Remove this line

**Review by Debskij — CHANGES_REQUESTED** [review]
Hi @karkadad! Thanks for your contribution. I've added some changes to change inference to single line approach.

**Review by ryanloney — APPROVED** [review]

**Review by Debskij — APPROVED** [review]

**Comment by ryanloney** [comment]
@karkadad there is another notebook called 216. Can you update this one? We also need to resolve the conflicts to hopefully trigger CI to run again

**Comment by karkadad** [comment]
@ryanloney @Debskij Thanks for reviewing the notebook. I've updated the notebook number and resolved the conflicts.

**Comment by karkadad** [comment]
@ryanloney @Debskij Looks like all the CI checks are passing, and also the notebook number is updated as well the conflicts are resolved now. Please review. Thanks

**Comment by ryanloney** [comment]
Thanks @karkadad! We will merge this tomorrow. @paularamo can you do a quick check and approve (unless you see any issues)?

**Review by paularamo — APPROVED** [review]
Thanks for your contributions. I will merge it, my concern in the future is with Win 11, but this is something we will work on in the future.

**Pull request merged**